app/controllers/Extractors.scala

```diff
@@ -80,6 +112,89 @@ class Extractors  @Inject() (extractions: ExtractionService,
+    for ((done, submitted) <- (myDonelist zip mySubmittedlist)) {
+      Logger.debug("my done date: " + done.start + ", fileid: " + done.file_id)
+      Logger.debug("my submitted date: " + submitted.start + ", fileid: " + submitted.file_id)
+      Logger.debug("done.start.getTime: " + done.start.getTime + ", submitted.start.getTime: " + submitted.start.getTime)
+      val diffInMillies = Math.abs(done.start.getTime - submitted.start.getTime)
+      Logger.debug("diffInMillies in ms: " + diffInMillies)
+      sum = sum + diffInMillies
+      n = n+1
+    }
+    sum = TimeUnit.SECONDS.convert(sum, TimeUnit.MILLISECONDS)
+    var average = BigDecimal.valueOf(sum)
+    Logger.debug("average: " + average)
+    if(n > 0) {
+      average = average/n
+    }
+    Logger.debug("average: " + average)
+    // get at the number of ``SUBMITTED'' submission in last week.
+    val lastweeksubmitted = extractions.findByExtractorIDBefore(extractorName, "SUBMITTED", last7daydate, 0)
+    Logger.debug("lastweek submitted: " + lastweeksubmitted.size)
+    // get at the number of ``SUBMITTED'' submission in last month.
+    val lastmonthsubmitted = extractions.findByExtractorIDBefore(extractorName, "SUBMITTED", lastmonthdate, 0)
+
+    // calculate the last 10 execution average time in the time range of last month between ``SUBMITTED'' and ``DONE''.
+    // get at most 10 ``DONE'' submission from last month.
+    val myLastTenDonelist = extractions.findByExtractorIDBefore(extractorName, "DONE", lastmonthdate, 10)
+    // get at most 10 ``SUBMITTED'' submission from last month.
+    val myLastTenSubmittedlist = extractions.findByExtractorIDBefore(extractorName, "SUBMITTED", lastmonthdate, 10)
+    n = 0
+    sum = 0
+    for ((done, submitted) <- (myLastTenDonelist zip myLastTenSubmittedlist)) {
+      Logger.debug("my last 10 done date: " + done.start + ", fileid: " + done.file_id)
+      Logger.debug("my last 10 submitted date: " + submitted.start + ", fileid: " + submitted.file_id)
+      Logger.debug("last 10 done.start.getTime: " + done.start.getTime + ", submitted.start.getTime: " + submitted.start.getTime)
+      val diffInMillies = Math.abs(done.start.getTime - submitted.start.getTime)
+      Logger.warn("last 10 diffInMillies in ms: " + diffInMillies)
+      sum = sum + diffInMillies
+      n = n+1
+    }
+    sum = TimeUnit.SECONDS.convert(sum, TimeUnit.MILLISECONDS)
+    var lastTenAverage = BigDecimal.valueOf(sum)
+    Logger.debug("last 10 average: " + lastTenAverage)
+    if(n > 0) {
+      lastTenAverage = lastTenAverage/n
+    }
+    Logger.warn("last 10 average: " + lastTenAverage)
+
+    val targetExtractor = extractorService.listExtractorsInfo(List.empty).find(p => p.name == extractorName)
+    targetExtractor match {
+      case Some(extractor) => Ok(views.html.extractorMetrics(extractorName, average.toString, lastTenAverage.toString, lastweeksubmitted.size, lastmonthsubmitted.size))
```

Comment:
This should count the number of unique jobs submitted, instead of the number of raw events we've seen:
```suggestion
      case Some(extractor) => Ok(views.html.extractorMetrics(extractorName, average.toString, lastTenAverage.toString, lastweeksubmitted, lastmonthsubmitted))
```